src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -835,28 +836,58 @@ export default class Fms {
      * @param routeString {string}
      * @return {boolean}
      */
-    replaceDepartureProcedure(routeString) {
-        const routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
+    replaceDepartureProcedure(routeString, airportIcao) {
+        const routeStringElements = routeString.toUpperCase().split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
 
-        if (routeStringElements.length !== 3) {
+        if (routeStringElements.length > 3) {
             return [false, 'departure procedure format not understood'];
         }
 
-        const procedureId = routeStringElements[1];
+        let procedureId = routeStringElements[0];
 
-        if (!NavigationLibrary.hasProcedure(procedureId)) {
+        if (routeStringElements.length === 3) {
```

Comment:
lets add a comment here too to make this cleaner.

we could use a `switch` on the length and then fire off smaller methods based on the length.  we can tackle that later, though

Reply:
added the comment.